include/flox/core/exceptions.hh

```diff
@@ -23,7 +24,9 @@ namespace flox {
 enum error_category {
   EC_OKAY                  = 0
 , EC_FAILURE               = 1
-, EC_PKG_QUERY_INVALID_ARG = 100
+, FLOX_EXCEPTION           = 100
```

Comment:
I think we need some explanation in the source here of what each category actually represents. It's unclear to me what the difference is between `EC_FAILURE` and `FLOX_EXCEPTION`.

Also, should all of the constants start with `EC_` for consistency, or is there some semantic difference you're trying to convey by some having that prefix and others not?

Reply:
Feel free to expand on the rough sketch I had there.

Really the only tips are "try to give yourself healthy breathing room of at least 99 sub-errors/category", and don't exceed `INT_MAX`

Reply:
Adding a comment and `EC_` prefixes